src/box/lua/console.lua

```diff
@@ -417,14 +420,68 @@ local function table_pack(...)
     return {n = select('#', ...), ...}
 end
 
+local initial_env
+
+-- Get initial console environment.
+--
+-- If the console_session_scope_vars compat option is set to new,
+-- each console session has its own variable scope (environment)
+-- created before execution of a first command.
+--
+-- This environment initially contains a couple of frequently used
+-- built-in modules. A non-local variable assignment within a
+-- command modifies the environment.
+--
+-- The initial environment is returned by console.initial_env().
+-- It is a table that may be modified by a user or completely
+-- replaced with a new table using console.set_initial_env().
+-- These calls don't affect existing console sessions, only ones
+-- that are created afterwards.
+function M.initial_env()
+    if initial_env == nil then
+        initial_env = {
+            clock     = require('clock'),
+            compat    = require('compat'),
+            config    = require('config'),
+            datetime  = require('datetime'),
+            decimal   = require('decimal'),
+            fiber     = require('fiber'),
+            fio       = require('fio'),
+            fun       = require('fun'),
+            json      = require('json'),
+            log       = require('log'),
+            msgpack   = require('msgpack'),
+            popen     = require('popen'),
+            uuid      = require('uuid'),
+            varbinary = require('varbinary'),
+            yaml      = require('yaml'),
+        }
+    end
+    return initial_env
+end
+
+-- Set initial console environment.
+--
+-- Accepts a table or a nil value. If the value is nil, the
+-- initial environment is reset to a default value.
+--
+-- See console.initial_env() for details.
+function M.set_initial_env(env)
+    if type(env) ~= 'table' and type(env) ~= 'nil' then
+        error(('console.set_initial_env: expected table or nil, got %s'):format(
+            type(env)), 0)
```

Comment:
Why is the error level 0 and not 2?

Reply:
That's my personal preference (we can't judge what level is a cause of the problem in the general case), but I don't mind to set it to 2. Changed.